executor/include/execution/executor/cuda_executor.hpp

```diff
@@ -11,16 +11,18 @@
 #include <execution/executor/base_executor.hpp>
 
 template <typename Interface, typename Cardinality, typename Blocking,
-    typename ProtoAllocator>
+          typename ProtoAllocator>
 struct cuda_executor;
 
 #ifdef CUDA
-template <> struct execution::executor_available<cuda_executor> : std::true_type {};
+template <>
+struct execution::executor_available<cuda_executor> : std::true_type {};
 #endif
 
-template <typename Interface,typename Cardinality,typename Blocking, typename ProtoAllocator>
-struct cuda_executor: executor<sse_executor, Interface, Cardinality, Blocking, ProtoAllocator> {
-
+template <typename Interface, typename Cardinality, typename Blocking,
```

Comment:
How about `<typename ...Args>` here too? It'll allow you to modify the executor class without worry of modifying the executors (just the usage, like in L44 and in consumer code)

Reply:
I need explicit access to the passed template arguments, like in the query functions in `executor` class. There will be some cases where I will need the argument name.